crates/rpc/rpc/src/otterscan.rs

```diff
@@ -198,10 +198,62 @@ where
     /// Handler for `getTransactionBySenderAndNonce`
     async fn get_transaction_by_sender_and_nonce(
         &self,
-        _sender: Address,
-        _nonce: u64,
-    ) -> RpcResult<Option<Transaction>> {
-        Err(internal_rpc_err("unimplemented"))
+        sender: Address,
+        nonce: u64,
+    ) -> RpcResult<Option<TxHash>> {
+        // Check if the sender is a contract
+        if self.has_code(sender, None).await? {
+            return Ok(None)
+        }
+
+        let highest =
+            self.eth.transaction_count(sender, None).await?.saturating_to::<u64>();
+
+        // If the nonce is higher or equal to the highest nonce, the transaction is pending or not
+        // exists.
+        if nonce >= highest {
+            return Ok(None)
+        }
+
+        // To find the first block where the sender's nonce is equal to the given one, we can use
+        // binary search within the range of blocks starting from block 1 up to the latest block
+        // number.
+        let mut low = 1;
+        let mut high = self.eth.block_number()?.saturating_to::<u64>();
+        let mut num = high;
+
+        while low <= high {
+            let mid = (low + high) / 2;
+            let mid_nonce = EthApiServer::transaction_count(
+                &self.eth,
+                sender,
+                Some(BlockId::Number(BlockNumberOrTag::Number(mid))),
+            )
+            .await?
+            .saturating_to::<u64>();
+
+            // The `transaction_count` returns the `nonce` after the transaction was executed, which
+            // is the state of the account after the block, and we need to find the
+            // transaction whose nonce is the pre-state, so need to compare with
+            // `nonce+1`.
+            if mid_nonce < nonce + 1 {
```

Comment:
```suggestion
            if mid_nonce == nonce + 1 {
                num = mid;
                break
            } else if mid_nonce <= nonce {
```

Reply:
there is no reason to keep looping when we have reached the desired condition

Reply:
Sorry, haven't changed it, I want the binary search to be more common, so the getContractCreator can reuse it. But before that I'll first add some tests

Reply:
> there is no reason to keep looping when we have reached the desired condition

I see what the problem is, if the nonce was changed in block 6, 10, so we get the same result in block 6, 7, 8, 9. so I think the condition `mid_nonce == nonce + 1` isn't match what we need(the mid can be any of 6..=9).